src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const transformedState = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties' :
+        Object.assign(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties' :
+        for (const key of action.keysToRemove) {
+          delete stateCopy[key];
+        }
+        break;
+
+      case 'clear' :
+        for (const key in stateCopy) {
+          delete stateCopy[key];
+        }
+        break;
+
+      default:
+        break;
```

Comment:
you need to throw some error here, so user can know, that he gave an unknown action type
(like `throw new Error('your error message')`)